### PR TITLE
Optimize images by using Netlify transformations and add portfolio content

### LINKED ISSUE #188: Use next/image for all images

### PR DESCRIPTION
**Pull request by KonradSzwarc**
Closes #188

### CODE REVIEW

File: src/utils/image.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+type TransformImageOptions = {
+  width: number;
+  height: number;
+  resize?: 'fit' | 'smartcrop';
+};
+
+export const transformImage = (imageSrc: string, { width, height, resize = 'fit' }: TransformImageOptions) =>
```
Comment (KonradSzwarc) [inline src/utils/image.ts:8]: We increase the requested image size by 1.2 factor as Netlify provides compression that lowers its quality. With this transformation, quality loss is less visible.

File: .gitignore [hunk 2]
```diff
@@ -31,3 +31,6 @@ yarn-error.log*
 
 # vercel
 .vercel
+
+# Local Netlify folder
+.netlify
```
Comment (mecies) [inline .gitignore]: Maybe add empty line to make github happy :D

File: .lfsconfig [hunk 3]
```diff
@@ -0,0 +1,2 @@
+[lfs]
+	url = https://cd2545dc-eb10-4a2a-a0c7-f9e2440e6bd2.netlify.app/.netlify/large-media
```
Comment (mecies) [inline .lfsconfig:2]: Is this indentation intended?

File: netlify.toml [hunk 4]
```diff
@@ -1,3 +1,6 @@
 [build]
-  command = "yarn build && yarn export"
-  publish = "out"
\ No newline at end of file
+  command = "yarn build"
+  publish = "out"
+
+[[plugins]]
+  package = "@netlify/plugin-nextjs"
```
Comment (mecies) [inline netlify.toml]: Also can we add empty line to make github happy? :D

File: package.json [hunk 5]
```diff
@@ -11,58 +11,58 @@
     "build:analyze": "cross-env ANALYZE=true yarn build",
```
Comment (mecies) [inline package.json:11]: How do you update all project dependencies?
`npm update` or `yarn upgrade` ?

File: src/components/Layout.tsx [hunk 6]
```diff
@@ -69,15 +69,18 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-export const Layout = ({ children, variant = 'default' }: LayoutProps) => {
+export const Layout = ({ children }: LayoutProps) => {
   const classes = useStyles();
+  const { pathname } = useRouter();
   const { componentType, isDesktop } = useComponentType();
 
+  const withResponsiveDetailsCard = !isDesktop && (pathname === '/' || pathname === '/contact');
```
Comment (mecies) [inline src/components/Layout.tsx:77]: Smart :D

File: src/utils/image.ts (hunk 1 above)
Comment (mecies) [reply src/utils/image.ts:8]: 🧙
Can you please add this comment in the code?

File: src/components/PortfolioCard/PortfolioCardDesktop.tsx [hunk 7]
```diff
@@ -70,6 +70,8 @@ export const PortfolioCardDesktop = ({
   const classes = useStyles();
   const [isHovered, setHovered] = useState(false);
 
+  const projectImage = transformImage(image, { width: 400, height: 224 });
```
Comment (mecies) [inline src/components/PortfolioCard/PortfolioCardDesktop.tsx:73]: Why previously you initialized values `width` and `height` as constants, and here you don't do it like so?

File: src/components/PortfolioCard/PortfolioCardTablet.tsx [hunk 8]
```diff
@@ -2,8 +2,13 @@ import React from 'react';
 import { Smartphone } from 'react-feather';
 import { Card, CardActionArea, CardContent, CardMedia, makeStyles, Typography } from '@material-ui/core';
 
+import { transformImage } from '@/utils/image';
+
 import type { PortfolioCardVariantProps } from './PortfolioCard.types';
 
+const PROJECT_IMAGE_WIDTH = 240;
```
Comment (mecies) [inline src/components/PortfolioCard/PortfolioCardTablet.tsx:9]: I'm not the biggest fan of declaring these two variables as constants, I think that it made sense when the values were the same for both height and width but here is seems redundant to me, wdyt?

**Review by mecies — DISMISSED** [review]
Looks good to me ✅ , I approve with some questions, but I didn't find any mistakes

File: .lfsconfig (hunk 3 above)
Comment (KonradSzwarc) [reply .lfsconfig:2]: This is an autogenerated file, so probably yes

File: package.json (hunk 5 above)
Comment (KonradSzwarc) [reply package.json:11]: `ncu -u && yarn`
I have [npm-check-updates](https://www.npmjs.com/package/npm-check-updates) installed

File: src/components/PortfolioCard/PortfolioCardTablet.tsx (hunk 8 above)
Comment (KonradSzwarc) [reply src/components/PortfolioCard/PortfolioCardTablet.tsx:9]: Those values need to be the same in the component and its styles, so this way we can ensure they always are

File: package.json (hunk 5 above)
Comment (mecies) [reply package.json:11]: Didn't know about this, thanks 😄

File: src/components/PortfolioCard/PortfolioCardDesktop.tsx (hunk 7 above)
Comment (KonradSzwarc) [reply src/components/PortfolioCard/PortfolioCardDesktop.tsx:73]: There was no need for that as those values are used only once here. In other cases, there were at least two places where I used sizing values.

File: src/components/PortfolioCard/PortfolioCardTablet.tsx (hunk 8 above)
Comment (mecies) [reply src/components/PortfolioCard/PortfolioCardTablet.tsx:9]: Makes sense 👍

File: src/components/PortfolioCard/PortfolioCardDesktop.tsx (hunk 7 above)
Comment (mecies) [reply src/components/PortfolioCard/PortfolioCardDesktop.tsx:73]: Makes sense, I would probably declare for consistency with other components but I agree that its not necessary

**Review by mecies — APPROVED** [review]

**Pull request merged**